dockerversion/useragent.go

```diff
@@ -8,8 +8,18 @@ import (
 )
 
 // DockerUserAgent is the User-Agent the Docker client uses to identify itself.
-// It is populated from version information of different components.
-func DockerUserAgent() string {
+// In accordance with RFC 7231 (5.5.3) is of the form:
+//    [upstream UA] [docker client's UA]
+// upstreamUA is optional
+func DockerUserAgent(upstreamUA ...string) string {
+	// ugliness to fake an optional arg in golang
+	var _upstreamUA string
+	if len(upstreamUA) == 0 {
+		_upstreamUA = ""
+	} else {
+		_upstreamUA = upstreamUA[0] + " upstream-ua/end; "
+	}
```

Comment:
As mentioned above, I would prefer to use a straight string argument for this function, but if you keep the current approach, this `if/else` block can be simplified to:

```go
 	if len(upstreamUA) > 0 {
 		_upstreamUA = upstreamUA[0] + " upstream-ua/end; "
 	}
```